src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+        this->unk298 += 1;
+
+        if (this->unk298 >= 8) {
+            this->unkTimer252 = Rand_ZeroFloat(30.0f);
+            this->unk298 = Rand_ZeroFloat(3.99000000954f);
+            // (3.99..) is loaded into f12 for rand_zero, but f12 never released
+            // b18:    jal     Rand_ZeroFloat              b18:    jal     Rand_ZeroFloat
+            // b1c:    lwc1    $f12,%lo(D_80893554)(at)  i b1c:    lwc1    $f12,%lo(.rodata+0x54)(at)
+            // b20:    lwc1    $f12,0x34(sp)             r b20:    lwc1    $f2,0x2c(sp)
+
+            if (newPos.y < 0.0f) {
+                newPos.y -= 100.0f;
+            } else {
+                newPos.y += 100.0f;
+            }
+            if (newPos.z < 0.0f) {
+                newPos.z -= 100.0f;
+            } else {
+                newPos.z += 100.0f;
+            }
+
+            this->unk2B0.x = this->unk2A4.x + newPos.y;
+            this->unk2B0.z = this->unk2A4.z + newPos.z;
+
+        } else {
+            this->unkTimer250 = 4;
+            if ((this->actor.bgCheckFlags & 1)) {
+                this->actor.speedXZ = 0.0f;
+                this->actor.velocity.y = 3.5f;
+            }
+        }
+    }
+
+    if (this->unkTimer250 != 0) {
+        Math_ApproachZeroF(&this->unk264[9], 0.5f, 4000.0f);
+        s16tmp = 1;
+        Math_ApproachF(&this->actor.world.pos.x, this->unk2B0.x, 1.0f, this->unk300);
+        Math_ApproachF(&this->actor.world.pos.z, this->unk2B0.z, 1.0f, this->unk300);
+        Math_ApproachF(&this->unk300, 3.0f, 1.0f, 0.3f);
+
+        newPos.x = this->unk2B0.x - this->actor.world.pos.x;
+        newPos.z = this->unk2B0.z - this->actor.world.pos.z;
+
+        if (fabsf(newPos.x) < 10.0f) {
+            newPos.x = 0.0;
+        }
+        if (fabsf(newPos.z) < 10.0f) {
+            newPos.z = 0.0;
+        }
+
+        if ((newPos.x == 0.0f) && (newPos.z == 0.0f)) {
+            this->unkTimer250 = 0;
+            this->unk298 = 7;
+        }
+
+        Math_SmoothStepToS(&this->actor.world.rot.y, Math_Atan2S(newPos.x, newPos.z), 3, this->unk304, 0);
+        Math_ApproachF(&this->unk304, 10000.0f, 1.0f, 1000.0f);
+    }
+    func_80891320(this, globalCtx, s16tmp);
+}
+#else
 #pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808919E8.asm")
+#endif
+
+void EnNiw_Held(EnNiw* this, GlobalContext* globalCtx) {
+    Vec3f vec3fcopy;
+    s16 rotZ;
+
+    // again with the double copy?
+    vec3fcopy = D_808934DC;
+    if (this->unkTimer250 == 0) {
+        this->unk29E = 2;
+        // cast req
+        this->unkTimer250 = (s32)(Rand_ZeroFloat(1.0f) * 10.0f) + 10;
+    }
+
+    // casts req
+    this->actor.shape.rot.x = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.x;
+    this->actor.shape.rot.y = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.y;
+    this->actor.shape.rot.z = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.z;
```

Comment:
```suggestion
    this->actor.shape.rot.x = (s16)randPlusMinusPoint5Scaled(5000.0f) + this->actor.world.rot.x;
    this->actor.shape.rot.y = (s16)randPlusMinusPoint5Scaled(5000.0f) + this->actor.world.rot.y;
    this->actor.shape.rot.z = (s16)randPlusMinusPoint5Scaled(5000.0f) + this->actor.world.rot.z;
```
There are many cases where you cast `s16` on a float so that the addition between two values is the addition of two `s16`, as ido will default to 2 floats if given 1 `f32` + 1 `s16`